### PR TITLE
hv & doc: nested VPID flush

### PR DESCRIPTION
**Pull request by ZideChen0**

### CODE REVIEW

File: doc/tutorials/nvmx_virtualization.rst [hunk 1]
```diff
@@ -71,6 +71,20 @@ hypervisor to actually run the L2 guest.
    a. L0 reflects most VMEXits to L1 hypervisor
    #. L0 runs L1 guest with VMCS01 and VMCS02 as the shadow VMCS
 
+VPID allocation
+===============
+
+ACRN doesn't emulate L2 VPIDs and allocates VPIDs for L1 VMs from the reserved top
+16-bit VPID range (`0x10000U - CONFIG_MAX_VM_NUM * MAX_VCPUS_PER_VM` and up).
```
Comment (dbkinder) [inline doc/tutorials/nvmx_virtualization.rst:78]:
```suggestion
16-bit VPID range (``0x10000U - CONFIG_MAX_VM_NUM * MAX_VCPUS_PER_VM`` and up).
```

File: doc/tutorials/nvmx_virtualization.rst [hunk 2]
```diff
@@ -71,6 +71,20 @@ hypervisor to actually run the L2 guest.
...
+If the L1 hypervisor enables VPID for L2 VMs and allocates L2 VPIDs not in this
+range, ACRN doesn't need to flush L2 VPID during L2 VMX transitions.
+
+This is the expected behavior in most of the time. But in special cases where a
+L2 VPID allocated by L1 hypervisor is within this reverved range, it's possible
```
Comment (dbkinder) [inline doc/tutorials/nvmx_virtualization.rst:83]:
```suggestion
L2 VPID allocated by L1 hypervisor is within this reserved range, it's possible
```

File: doc/tutorials/nvmx_virtualization.rst [hunk 3]
```diff
@@ -71,6 +71,20 @@ hypervisor to actually run the L2 guest.
...
+that this L2 VPID may conflict with a L1 VPID. In this case,  ACRN flushes VPID
+on L2 VMExit/VMEntry that are associated with this L2 VPID, which may significantly
+impact performances of this L2 VM.
```
Comment (dbkinder) [inline doc/tutorials/nvmx_virtualization.rst:86]:
```suggestion
and negatively impact performance of this L2 VM.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
clarity, spelling, and formatting fixes

File: doc/tutorials/nvmx_virtualization.rst (hunk 1 above)
Comment (ZideChen0) [reply doc/tutorials/nvmx_virtualization.rst:78]: @dbkinder thank you very much for your comments! I will get them fixed.

**Review by dbkinder — APPROVED** [review]
Doc changes LTGM, thanks.  (Still need to fix CI check failures.)

**Comment by dbkinder** [comment]
@ZideChen0 Try adding Tracked-On: #6289 to both commits to get rid of the Tracked-On CI failure

**Comment by wenlingz** [comment]
@dongyaozu could you please help approve this PR , thanks

**Review by dongyaozu — CHANGES_REQUESTED** [review]

File: doc/tutorials/nvmx_virtualization.rst [hunk 4]
```diff
@@ -71,6 +71,20 @@ hypervisor to actually run the L2 guest.
...
+ACRN doesn't emulate L2 VPIDs and allocates VPIDs for L1 VMs from the reserved top
+16-bit VPID range (``0x10000U - CONFIG_MAX_VM_NUM * MAX_VCPUS_PER_VM`` and up).
+If the L1 hypervisor enables VPID for L2 VMs and allocates L2 VPIDs not in this
+range, ACRN doesn't need to flush L2 VPID during L2 VMX transitions.
+
+This is the expected behavior in most of the time. But in special cases where a
+L2 VPID allocated by L1 hypervisor is within this reserved range, it's possible
+that this L2 VPID may conflict with a L1 VPID. In this case,  ACRN flushes VPID
+on L2 VMExit/VMEntry that are associated with this L2 VPID, which may significantly
+negatively impact performances of this L2 VM.
```
[the remark this answers is not in the document]
Comment (ZideChen0) [reply doc/tutorials/nvmx_virtualization.rst:86]: > We need to highlight this limitation to someplace that the developers/customers won't miss. Do we have any chapters that the developers must read and remember in mind?

This doc has a "Restrictions and Constraints" section. I can move this VPID part to there.

**Review by dongyaozu — APPROVED** [review]

**Pull request merged**